### ISSUE TITLE
[Bug] Including {datetime} in template on Windows results in OSError

### ISSUE THREAD

**Issue by kuchenmitsahne** [althonos/InstaLooter#22]
Since the `{datetime}` template option puts the date and time in the format `YYYY-MM-DD hh:mm:ss`, any attempt to open a file for writing with that option on Windows will invariably fail because the colon is a reserved character in Windows filenames.  Here's an example traceback from a command line invocation run with `-T {datetime}`:

```
Exception in thread Thread-3:
Traceback (most recent call last):
  File "c:\python35-32\lib\threading.py", line 914, in _bootstrap_inner
    self.run()
  File "c:\python35-32\lib\site-packages\instaLooter\worker.py", line 41, in run
    self._download_photo(media)
  File "c:\python35-32\lib\site-packages\instaLooter\worker.py", line 85, in _download_photo
    self._dl(photo_url, photo_name)
  File "c:\python35-32\lib\site-packages\instaLooter\worker.py", line 108, in _dl
    with open(dest, 'wb') as dest_file:
OSError: [Errno 22] Invalid argument: '.\\2017-01-12 16:14:41.jpg'
```

(there are similar exceptions in the other threads)

I did a hacky fix so I could use the program as desired by simply appending `.replace(':', '.')` to the calls to `_make_filename()` in `worker.py`. I've had no issues since, so I'm fairly confident this was the only problem.

I'm a novice, so I'm not quite sure what the proper fix ought to be (adding another template option that explicitly doesn't use colons or changing the current template, detecting the OS and silently swapping the colons for an alternate character, something else entirely).

**Comment by althonos** [comment]
Thank you very much for that issue, I admit I'm not thoroughly testing against Windows. I'll update the datetime rendering to do something like `2017-01-20 16h19m05s`, that shouldn't trigger Windows anymore.

**Comment by althonos** [comment]
Fixed in d48b60abb72b6ec2a87a6eefd9b1bf07a9e5738d with additional acceptance tests. `{datetime}` now generates a string following format `2017-01-20 16h19m05s356754` (microseconds added for uniqueness).
